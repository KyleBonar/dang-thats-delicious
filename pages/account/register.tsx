import * as React from "react";
import RegisterUser from "../../src/components/RegisterUser/RegisterUser";
import { Article } from "../../src/components/Article/Article";
import HeaderSimple from "../../src/components/HeaderSimple/HeaderSimple";

export interface RegisterProps {}

const Register: React.SFC<RegisterProps> = () => {
  return (
    <>
      <HeaderSimple />
      <Article size="sm">
        <RegisterUser />
      </Article>
    </>
  );
};

export default Register;
