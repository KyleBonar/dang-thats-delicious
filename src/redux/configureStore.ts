import { createStore, applyMiddleware, combineReducers, Action } from "redux";
import { composeWithDevTools } from "redux-devtools-extension/developmentOnly";
import thunk, { ThunkAction, ThunkDispatch } from "redux-thunk";

import flashMessage from "./flashMessage/reducer";
import sauces from "./sauces/reducer";
import reviews from "./reviews/reducer";
import users from "./users/reducer";
import { ISauce } from "./sauces/types";
import { IQuery } from "./sauces/types";
import { IUser } from "./users/types";
import { IReview } from "./reviews/types";
import { IFlashState } from "./flashMessage/types";
import Auth from "../utils/Auth/Auth";

const rootReducer = combineReducers({
  flashMessage,
  sauces,
  reviews,
  users
});

export interface AppState {
  sauces: {
    allSlugs?: string[];
    bySlug?: { [key: string]: ISauce };
    total?: number;
    query?: IQuery;
    types: string[];
    saucesWithNewestReviews: [];
    newest: [];
    featured: [];
  };
  users: {
    self: { token?: string; displayName?: string; avatar?: string };
    byDisplayName?: { [key: string]: IUser };
    allDisplayNames?: string[];
  };
  reviews: {
    byHashID?: { [key: string]: IReview };
    allHashIDs?: string[];
  };
  flashMessage: IFlashState;
}

export const configureStore = () => {
  // was of type 'object'
  const initialState: AppState = {
    sauces: {
      allSlugs: [],
      bySlug: {},
      total: 0,
      query: {},
      types: [
        "Hot Sauce",
        "BBQ Sauce",
        "Gravy",
        "Marinade",
        "Salsa",
        "Meat Sauce"
      ],
      saucesWithNewestReviews: [],
      newest: [],
      featured: []
    },
    users: {
      self: {
        token: Auth.isUserAuthenticated() ? Auth.getToken() : undefined,
        displayName: Auth.isUserAuthenticated ? Auth.getName() : undefined
      },
      byDisplayName: {},
      allDisplayNames: []
    },
    reviews: { byHashID: {}, allHashIDs: [] },
    flashMessage: { isVisible: false, type: null, text: null, slug: null }
  };

  if (process.env.NODE_ENV === "production") {
    return createStore(rootReducer, initialState, applyMiddleware(thunk));
  } else {
    return createStore(
      rootReducer,
      initialState,
      composeWithDevTools(applyMiddleware(thunk))
    );
  }
};

// Defining thunk properties
export type MyThunkResult<R> = ThunkAction<R, AppState, undefined, Action>;
// It is important to use Action as last type argument, does not work with any.
export type MyThunkDispatch = ThunkDispatch<AppState, undefined, Action>;
