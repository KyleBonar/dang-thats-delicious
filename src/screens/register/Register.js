import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";

import RegisterForm from "./components/registerForm/RegisterForm";

import { flashError, flashClose } from "../../redux/actions/flash";
import { register } from "../../redux/actions/auth";

class Register extends Component {
  static propType = {
    history: PropTypes.shape({ push: PropTypes.func.isRequired }).isRequired,
    register: PropTypes.func.isRequired,
    flashError: PropTypes.func.isRequired,
    flashClose: PropTypes.func.isRequired
  };
  render() {
    return (
      <div className="inner">
        <RegisterForm onSubmit={this.handleSubmit} />
      </div>
    );
  }

  handleSubmit = e => {
    const data = {
      user: {
        email: e.email,
        name: e.name,
        password: e.password,
        confirmPassword: e.confirmPassword
      }
    };
    this.props.flashClose();
    this.props
      .register(data)
      .then(() => this.props.history.push("/sauces"))
      .catch(err => {
        console.log(err);
        console.log(err.response);
        this.props.flashError({ text: err.response.data.msg });
      });
  };
}

const mapDispatchToProps = { register, flashError, flashClose };

export default connect(null, mapDispatchToProps)(Register);
