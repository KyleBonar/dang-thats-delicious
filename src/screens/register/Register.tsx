import * as React from "react";
import { connect } from "react-redux";
import LogoSFYT from "../../images/icons/LogoSFYT";
import styled from "styled-components";
import Article from "../../components/Article/Article";
import PageTitle from "../../components/PageTitle/PageTitle";
import TextInput from "../../components/TextInput/TextInput";

const StyledLogoContainer = styled.div`
  max-width: 150px;
  margin: 0 auto;
  padding: 1em;
`;

const StyledArticle = styled(Article)`
  max-width: 600px;
`;
const StyledFormContainer = styled.div`
  background-color: ${props => props.theme.formContainerBackgroundColor};
  padding: 1.5rem;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
`;

export interface RegisterProps {}

export interface RegisterState {}

class Register extends React.Component<RegisterProps, any> {
  public render() {
    return (
      <div>
        <StyledLogoContainer>
          <LogoSFYT />
        </StyledLogoContainer>
        <hr />
        <StyledArticle>
          <PageTitle>Register</PageTitle>
          <StyledFormContainer>
            <TextInput
              type="text"
              onChange={() => {}}
              showLabel={true}
              label={"Email"}
              required={true}
            />
            <TextInput
              type="text"
              onChange={() => {}}
              showLabel={true}
              label={"Confirm Email"}
              required={true}
            />
            <TextInput
              type="text"
              onChange={() => {}}
              showLabel={true}
              label={"Password"}
              required={true}
            />
            <TextInput
              type="text"
              onChange={() => {}}
              showLabel={true}
              label={"Confirm Password"}
              required={true}
            />
            <TextInput
              type="text"
              onChange={() => {}}
              showLabel={true}
              label={"Display Name"}
              required={true}
            />
          </StyledFormContainer>
        </StyledArticle>
      </div>
    );
  }
}

const mapState2Props = state => {
  return {};
};

export default connect(mapState2Props)(Register);
