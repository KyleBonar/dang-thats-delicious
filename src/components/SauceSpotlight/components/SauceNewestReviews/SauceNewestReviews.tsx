import * as React from "react";

import List from "../../../List/List";
import { useSelector } from "react-redux";
import { AppState } from "../../../../redux/configureStore";

interface ISauceNewestReviewsProps {}

const SauceNewestReviews: React.FunctionComponent<ISauceNewestReviewsProps> = props => {
  // defaults
  const _loadingTxt = "loading...";
  const _noNewSauces = "Could not find any new sauces!";
  const _title = "Newly Reviewed";

  const sauces = useSelector(
    (store: AppState) => store.sauces.saucesWithNewestReviews
  );

  return (
    <>
      {!sauces || !sauces || sauces.length === 0 ? (
        <p>{_noNewSauces}</p>
      ) : (
        <List
          items={sauces.map((x, ind) => {
            return {
              link: `/sauce/view?s=${x.slug}`,
              text: x.name,
              id: `${ind}-${x.name}`
            };
          })}
          title={_title}
        />
      )}
    </>
  );
};

export default SauceNewestReviews;
