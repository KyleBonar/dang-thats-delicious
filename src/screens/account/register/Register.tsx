import * as React from "react";
import { connect } from "react-redux";

import { register } from "../../../redux/users/actions";
import { IRegisterUser } from "../../../redux/users/types";
import LogoSFYT from "../../../images/icons/LogoSFYT";
import PageTitle from "../../../components/PageTitle/PageTitle";
import { TextInput } from "../../../components/TextInput/TextInput";
import { Link } from "../../../components/Link/Link";
import { AppState } from "../../../redux/configureStore";
import {
  StyledDiv,
  StyledLogoContainer,
  StyledArticle,
  StyledFormContainer,
  StyledText,
  StyledButton
} from "./RegisterStyle";
import {
  FlashMessage,
  FlashMessageProps
} from "../../../components/FlashMessage/FlashMessage";
import Auth from "../../../utils/Auth/Auth";

export interface RegisterProps {
  register: any;
  history: { push: (location: string) => null };
}

export interface RegisterState {
  email: string;
  confirmEmail: string;
  password: string;
  confirmPassword: string;
  displayName: string;
  flashMessage: FlashMessageProps;
}

class Register extends React.Component<RegisterProps, RegisterState> {
  constructor(props: RegisterProps) {
    super(props);

    // Init state
    this.state = {
      email: "",
      confirmEmail: "",
      password: "",
      confirmPassword: "",
      displayName: "",
      flashMessage: {
        isVisible: false
      }
    };
  }

  public render() {
    return (
      <StyledDiv>
        <StyledLogoContainer>
          <Link to="/">
            <LogoSFYT />
          </Link>
        </StyledLogoContainer>
        <hr />
        <StyledArticle>
          <PageTitle>Register</PageTitle>
          <StyledFormContainer>
            {this.state.flashMessage.isVisible && (
              <FlashMessage type={this.state.flashMessage.type} isVisible>
                {this.state.flashMessage.text}
              </FlashMessage>
            )}
            <form onSubmit={this.onSubmit} style={{ width: "100%" }}>
              <TextInput
                type="email"
                onChange={this.onTextChange}
                showLabel={true}
                label={"Email"}
                name={"email"}
                value={this.state.email}
                required={true}
              />
              <TextInput
                type="email"
                onChange={this.onTextChange}
                showLabel={true}
                label={"Confirm Email"}
                name={"confirmEmail"}
                value={this.state.confirmEmail}
                required={true}
                requirementText={"Must match above."}
              />
              <TextInput
                type="password"
                onChange={this.onTextChange}
                showLabel={true}
                label={"Password"}
                name={"password"}
                value={this.state.password}
                required={true}
                requirementText={"Must be at least 9 characters long."}
              />
              <TextInput
                type="password"
                onChange={this.onTextChange}
                showLabel={true}
                label={"Confirm Password"}
                name={"confirmPassword"}
                value={this.state.confirmPassword}
                required={true}
                requirementText={"Must match above."}
              />
              <TextInput
                type="text"
                onChange={this.onTextChange}
                showLabel={true}
                label={"Display Name"}
                name={"displayName"}
                value={this.state.displayName}
                required={true}
                requirementText={"Must be at least 6 characters long."}
              />
              <StyledText>
                By clicking 'Register', you agree to Sauce For Your Thoughts{" "}
                <Link to="#">Terms and Conditions</Link>
              </StyledText>
              <StyledButton type="submit">Register</StyledButton>
            </form>
          </StyledFormContainer>
        </StyledArticle>
      </StyledDiv>
    );
  }

  private onTextChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
    if (!event || !event.target) {
      return;
    }
    // Grab the name and value
    const { name, value }: { name: string; value: string } = event.target;

    // Update local state
    this.setState({
      ...this.state,
      [name]: value
    });
  };

  private onSubmit = async (event: React.FormEvent): Promise<any> => {
    event.preventDefault();

    if (this.state.email !== this.state.confirmEmail) {
      this.setState({
        flashMessage: {
          isVisible: true,
          text: "Your emails do not match. Please fix this before continuing.",
          type: "alert"
        }
      });

      window.scrollTo(0, 0); // Move screen to top

      return;
    }

    if (this.state.password !== this.state.confirmPassword) {
      this.setState({
        flashMessage: {
          isVisible: true,
          text:
            "Your passwords do not match. Please fix this before continuing.",
          type: "alert"
        }
      });

      window.scrollTo(0, 0); // Move screen to top

      return;
    }

    if (this.state.password.length < 8) {
      this.setState({
        flashMessage: {
          isVisible: true,
          text:
            "Your password is too weak! Please make your password over 8 characters long.",
          type: "alert"
        }
      });

      window.scrollTo(0, 0); // Move screen to top

      return;
    }

    if (this.state.displayName.length < 6) {
      this.setState({
        flashMessage: {
          isVisible: true,
          text:
            "Your display name is not long enough. It must be at least 6 characters long.",
          type: "alert"
        }
      });

      window.scrollTo(0, 0); // Move screen to top

      return;
    }

    // Grab values from state
    const {
      email,
      confirmEmail,
      password,
      confirmPassword,
      displayName
    } = this.state;

    // Create credentials obj
    const credentials: IRegisterUser = {
      user: { email, confirmEmail, password, confirmPassword, displayName }
    };
    try {
      // dispatch action which calls API to register user
      await this.props.register({ credentials });

      // Redirect user to sauces page -- Maybe take them to user home page instead?
      this.props.history.push("/sauces");
    } catch (err) {
      // Create warning flash
      this.setState({
        flashMessage: {
          isVisible: true,
          text: err.response.data.msg,
          type: "warning"
        }
      });

      window.scrollTo(0, 0); // Move screen to top
    }
  };
}

const mapState2Props = (state: AppState) => {
  return {};
};

const mapDispatch2Props = {
  register
};

export default connect(
  mapState2Props,
  mapDispatch2Props
)(Register);
