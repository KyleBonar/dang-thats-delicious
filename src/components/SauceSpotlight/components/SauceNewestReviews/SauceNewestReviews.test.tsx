import "jsdom-global/register";
import * as React from "react";
import * as enzyme from "enzyme";
import { Provider } from "react-redux";
import { MockStoreEnhanced } from "redux-mock-store";
import { ReactWrapper } from "enzyme";

import {
  fakeStore,
  ITERATION_SIZE
} from "../../../../utils/testUtils/testUtils";

import SauceNewestReviews from "./SauceNewestReviews";
import { AppState } from "../../../../redux/configureStore";

describe("<SauceNewestReviews />", () => {
  // defaults
  const _noNewSauces = "Could not find any new sauces!";
  const _title = "Newly Reviewed";

  let wrappers: ReactWrapper<
    any,
    Readonly<{}>,
    React.Component<{}, {}, any>
  >[] = [];
  let fakeStores: MockStoreEnhanced<unknown, {}>[] = [];

  beforeAll(() => {
    fakeStores = new Array(ITERATION_SIZE).fill(null).map(fakeStore);

    wrappers = new Array(ITERATION_SIZE).fill(null).map((x, ind) => {
      return enzyme.mount(
        <Provider store={fakeStores[ind]}>
          <SauceNewestReviews />
        </Provider>
      );
    });
  });

  it("renders", () => {
    wrappers.forEach(wrapper => {
      expect(wrapper.exists()).toBeTruthy();
    });
  });

  it("matches snapshot", () => {
    wrappers.forEach(wrapper => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  it("renders default empty text if no sauces found", () => {
    wrappers.forEach((wrapper, ind) => {
      const reduxStore = fakeStores[ind].getState() as AppState;

      // if we have newest sauces, then we wont render out the default so we can skip
      if (reduxStore.sauces.newest && reduxStore.sauces.newest.length > 0)
        return;

      expect(wrapper.text()).toContain(_noNewSauces);
    });
  });

  it("renders a List component if sauces found", () => {
    wrappers.forEach((wrapper, ind) => {
      const reduxStore = fakeStores[ind].getState() as AppState;

      // Make sure we have sauces to render
      if (!reduxStore.sauces.newest) return;
      if (reduxStore.sauces.newest.length === 0) return;

      expect(wrapper.find("List").exists()).toBeTruthy();
    });
  });
});
