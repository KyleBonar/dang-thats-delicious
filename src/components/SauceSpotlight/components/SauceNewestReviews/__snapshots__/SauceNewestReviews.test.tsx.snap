// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SauceNewestReviews /> matches snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 2`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 3`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 4`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 5`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 6`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 7`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 8`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 9`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 10`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 11`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 12`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 13`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 14`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 15`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 16`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 17`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 18`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 19`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 20`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 21`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 22`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 23`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 24`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 25`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 26`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 27`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 28`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 29`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 30`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 31`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;

exports[`<SauceNewestReviews /> matches snapshot 32`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SauceNewestReviews>
    <p>
      Could not find any new sauces!
    </p>
  </SauceNewestReviews>
</Provider>
`;
